docs/docs/user-guide.md

```diff
@@ -4,115 +4,141 @@ sidebar_position: 2
-haven't said whether `x` can be null or not, we don't know whether
-`x.getClass()` is safe (at least as far as JSpecify is concerned).
+There are four JSpecify annotations that are used together to indicate the
+nullness of all symbols.
 
-There are two JSpecify annotations that indicate these properties:
+*   [`@Nullable`] applied to a type means a value of that type can be `null`.
+    `@Nullable String x` means that `x` might be `null`.
 
-*   `@Nullable` applied to a type means a reference of that type can be null.
+*   [`@NonNull`] applied to a type means a value of that type cannot be `null`.
+    `@NonNull String x` means that `x` is never `null` (or you have a bug in
+    your code).
 
-*   `@NullMarked` applied to a module, package, or class means that a reference
-    in that scope can't be null unless its type is explicitly marked
+*   [`@NullMarked`] applied to a module, package, class, or method means that a
+    value in that scope can't be `null` unless its type is explicitly marked
     `@Nullable`. (Below we will see that there are some exceptions to this for
     [local variables](#local-variables) and
-    [type variables](#defining-generics).)
+    [type variables](#declaring-generics).) In code covered by `@NullMarked`,
+    `String x` means the same as `@NonNull String x`.
 
-The notion of "can't be null" should really be read with a footnote that says
+*   [`@NullUnmarked`] applied to a package, class, or method undoes the effects
+    of any surrounding `@NullMarked`. Values in its scope (unless counteracted
+    by an enclosed `@NullMarked` generally have unspecified nullness unless they
+    are annotated with `@Nullable` or `@NonNull`, as if there were no enclosing
+    `@NullMarked` at all.
+
+The notion of "can't be `null`" should really be read with a footnote that says
 "if all the code in question is `@NullMarked`". For example, if you have some
 code that is not `@NullMarked` and that calls a `@NullMarked` method, then tools
-might allow it to pass a possibly-null value to a method that is expecting a
-"can't be null" parameter.
+might allow it to pass a possibly-`null` value to a method that is expecting a
+"can't be `null`" parameter.
+
+## `@Nullable` and `@NonNull`
 
-## `@Nullable`
+The [`@Nullable`] annotation applied to a type means that a value of the type
+can be `null`. Code that uses those values must be able to deal with the `null`
+case, and it's okay to assign `null` to those values or pass `null` to those
+parameters.
 
-The `@Nullable` annotation applied to a type means that that use of the type
-includes references that can be null. Code that deals with those references must
-be able to deal with the null case.
+The [`@NonNull`] annotation applied to a type means that no value of the type
+can be `null` (unless there's a bug in your code). Code that uses those values
+can assume they're not `null`, but it's a bad idea to assign `null` to those
```

Comment:
I'm contrasting "you _can_ do this, but you _shouldn't_ do that".

Reply:
I agree that "but" is weird: There's a contrast going on because what you can and can't/shouldn't do, but I'd compare it to "My favorite color is red, but it's not blue." In my reading, it would work better with "so" or "and" than with "but." "Accordingly, code that uses those values... and it's a bad idea...."